### PR TITLE
Zadanie 3 - ozio1212

### PR DESCRIPTION
**Pull request by ozio1212**

### CODE REVIEW

**Review by MrChazar — APPROVED** [review]
Uwagi:
- dobrze opisana praca, wszystko jest czytelne i na miejscu
- dobre wykorzystanie sieci cnn
Git praca

**Review by Hanzoze — COMMENTED** [review]
Dobrze napisana praca. Wypisano wnioski z poprzednich zajęć które mogą pomóc w tym ćwiczeniu.  Są wnioski na końcu zadania i dobrze zauważone, że na MNIST wystarcza proste CNN, dlatego różnica Baseline i CNN to już jest sukces 🙏

**Review by rozamazurek — COMMENTED** [review]
Super praca, bardzo fajnie poprowadzony notebook, wszystko opisane, przejrzyście się to pisało. Fajnie, że z każdym punktem, wyniki modelu były lepsze, spoko analiza z dodaniem batch norm i dropout.

**Comment by GregW04** [comment]
@ozio1212  Cześć przychodzę z code review:
- podział na train-val-test + 
- Baseline + 
- CNN +
- BatchNorm + 
- Dropout + 
-  krzywe uczenia + 
- kod czytelny +
- brakuje mi trochę pokazanego eksperymentu (krok po kroku ulepszanie sieci itd. chodzi mi o kroki jeszcze przed BatchNorm, i Dropout) --
- brak stricte rozszerzania -
- brak metryk Recall, Precision, F1-score, ROC-AUC - 
- wniosek troche za krotki, ale ok! +
-  Co bym dodał:
- minusy z powyżej 
- eksperymenty na rożnych optimizerach i learning-rate'ach
- wgłębienie się w problem macierz pomyłek i które liczby mylą modele
- eksperymenty na różnych funkcjach aktywacji 
- wykresy słupkowe(metryka i loss) dla znacznych epok (1 i ostatnia) dla różnych opcji modeli
- Oglonie praca poprawna i czytelna ,ale nie widzę abyś zbyt eksperymentował/wgłębił się w problem +
Dzięki
